### PR TITLE
layers: Remove extra SHADER_MODULE_STATE constructor

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Reduce the number of `SHADER_MODULE_STATE` constructors to only what is needed

### CODE REVIEW

File: layers/debug_printf.cpp [hunk 1]
```diff
@@ -240,7 +240,7 @@ std::vector<DPFSubstring> DebugPrintf::ParseFormatString(const std::string &form
 
 std::string DebugPrintf::FindFormatString(std::vector<uint32_t> pgm, uint32_t string_id) {
     std::string format_string;
-    SHADER_MODULE_STATE module_state(pgm);
+    SHADER_MODULE_STATE module_state(pgm.data(), pgm.size() * sizeof(uint32_t));
```
Comment (juan-lunarg) [inline layers/debug_printf.cpp:243]: The constructor would be more robust if it took a `layer_data::span` instead of an individual pointer / size.

**Review by jeremyg-lunarg — DISMISSED** [review]
LGTM

**Review by ncesario-lunarg — DISMISSED** [review]

File: layers/gpu_validation/gpu_validation.cpp [hunk 2]
```diff
@@ -1149,7 +1149,7 @@ void GpuAssisted::AnalyzeAndGenerateMessages(VkCommandBuffer command_buffer, VkQ
     std::string vuid_msg;
     VkShaderModule shader_module_handle = VK_NULL_HANDLE;
     VkPipeline pipeline_handle = VK_NULL_HANDLE;
-    std::vector<uint32_t> pgm;
+    layer_data::span<const uint32_t> pgm;
```
Comment (juan-lunarg) [inline layers/gpu_validation/gpu_validation.cpp:1152]: Avoids copy on line 1161

File: layers/gpu_validation/debug_printf.cpp [hunk 3]
```diff
@@ -313,7 +313,7 @@ void DebugPrintf::AnalyzeAndGenerateMessages(VkCommandBuffer command_buffer, VkQ
         std::stringstream shader_message;
         VkShaderModule shader_module_handle = VK_NULL_HANDLE;
         VkPipeline pipeline_handle = VK_NULL_HANDLE;
-        std::vector<uint32_t> pgm;
+        layer_data::span<uint32_t> pgm;
```
Comment (juan-lunarg) [inline layers/gpu_validation/debug_printf.cpp:316]: avoids copy of line 325

File: layers/debug_printf.cpp (hunk 1 above)
Comment (juan-lunarg) [reply layers/debug_printf.cpp]: Went ahead and update this PR to use `layer_data::span` which led me to find std::vector copies that could be eliminated with span usage

**Review by juan-lunarg — APPROVED** [review]

**Pull request merged**